DuckDuckGo/PassKitPreviewHelper.swift

```diff
@@ -42,3 +44,45 @@ class PassKitPreviewHelper: FilePreview {
         }
     }
 }
+
+class ZippedPassKitPreviewHelper: FilePreview {
+    private weak var viewController: UIViewController?
+    private let filePath: URL
+    
+    required init(_ filePath: URL, viewController: UIViewController) {
+        self.filePath = filePath
+        self.viewController = viewController
+    }
+    
+    func preview() {
+        if let passes: [PKPass] = extractDataFromZip(at: self.filePath)?.compactMap({ try? PKPass(data: $0) }),
+           passes.count > 0,
+           let controller = PKAddPassesViewController(passes: passes) {
+            viewController?.present(controller, animated: true)
+        } else {
+            os_log("Can't present passkit: No passes in passes file", type: .debug)
+        }
+    }
+ 
+    func extractDataFromZip(at zipPath: URL) -> [Data]? {
+        var dataObjects = [Data]()
+        do {
+            let archive = try Archive(url: zipPath, accessMode: .read)
+            try archive.forEach { entry in
+                var passData = Data()
+                _ = try archive.extract(entry, skipCRC32: true) { data in
+                    passData.append(data)
+                }
+                
+                if passData.count > 0 {
+                    dataObjects.append(passData)
+                }
+            }
+        } catch {
+            os_log("Error reading pkpasses file: %s", type: .debug, error.localizedDescription)
```

Comment:
Same here, why logging error as .debug and not .error?

Reply:
following what was already in the file. will update them all